File: MechanicalSheets.Api/Models/DefectCatalog.cs
using MechanicalSheets.Api.Enums;

namespace MechanicalSheets.Api.Models;


public class DefectCatalog
{
    public int Id { get; set; }
    public string Code { get; set; } = string.Empty;    
    public DefectCategoryEnum Category { get; set; } = DefectCategoryEnum.Bodywork;   
    public string Description { get; set; } = string.Empty;
    public byte Gravity { get; set; }

    public ICollection<SheetDefectItem> DefectItems { get; set; } = new List<SheetDefectItem>();
}

File: frontend-angular/src/app/models/defect-catalog.ts
// Specchio di DefectCategoryEnum.cs
export type DefectCategory = 'Bodywork' | 'Lighting' | 'Fasteners' | 'Frame';

// GET /api/defect-catalog → {Id, Code, Category, Description, Gravity}
export interface DefectCatalog {
  id: number;
  code: string;
  category: DefectCategory;
  description: string;
  gravity: number;
}
